### PR TITLE
Enable spellcheck for `ipld/`

### LINKED ISSUE #1746: Enable `cargo-spellcheck` for `ipld/`
**Issue summary**

Make sure that `cargo spellcheck --cfg ../.config/spellcheck.toml` can run without errors in `ipld/`. For details, see the tracking issue.

**Other information and links**

### PR DESCRIPTION
**Pull request by connormullett**
**Summary of changes**
Changes introduced in this pull request:
- Spellcheck `ipld/`

**Reference issue to close (if applicable)**

Closes #1746 

**Other information and links**

### CODE REVIEW

File: .config/forest.dic [hunk 1]
```diff
@@ -11,7 +11,7 @@ Filecoin
 IPLD
 JSON
 JWT
-Merkle
+merkle
```
Comment (jdjaustin) [inline .config/forest.dic:14]: In most cases I think `Merkle` is correct (capitalized, it's someone's name I think).

**Review by jdjaustin — COMMENTED** [review]
``

File: ipld/src/selector/mod.rs [hunk 2]
```diff
@@ -168,12 +168,12 @@ impl SubAssign<u64> for RecursionLimit {
...
-///   - in Matcher, to determine if a node is selected.
-///   - in ExploreRecursive, to halt exploration.
-///   - in ExploreConditional,
+///   - in `Matcher`, to determine if a node is selected.
+///   - in `ExploreRecursive`, to halt exploration.
+///   - in `ExploreConditional`,
 ///
 ///
-/// TODO -- Condition is very skeletal and incomplete.
+/// TO-DO: Condition is very skeletal and incomplete.
```
Comment (lemmih) [inline ipld/src/selector/mod.rs:176]: Move this `TODO` out of the documentation.

File: .config/forest.dic [hunk 3]
```diff
@@ -33,3 +33,9 @@ tuple
 validator/S
 mempool
 sortition
+struct/S
```
Comment (lemmih) [inline .config/forest.dic:36]: Let's type it out as `structure`.

File: .config/forest.dic [hunk 4]
```diff
@@ -33,3 +33,9 @@ tuple
...
+serializer/SM
+deserialize/DG
+deserializer/SM
+boolean
```
Comment (lemmih) [inline .config/forest.dic:40]: `boolean` is already in the dictionary.

File: .config/forest.dic [hunk 5]
```diff
@@ -33,3 +33,9 @@ tuple
 validator/S
 mempool
 sortition
+struct/S
+serializer/SM
```
Comment (lemmih) [inline .config/forest.dic:37]: `serializer` is already in the dictionary.

File: .config/forest.dic [hunk 6]
```diff
@@ -33,3 +33,9 @@ tuple
...
+deserialize/DG
+deserializer/SM
+boolean
+baz
```
Comment (lemmih) [inline .config/forest.dic:41]: What does `baz` mean?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: .config/forest.dic (hunk 1 above)
Comment (connormullett) [reply .config/forest.dic]: That makes, fixed up

File: .config/forest.dic (hunk 3 above)
Comment (connormullett) [reply .config/forest.dic]: 👍

File: ipld/src/selector/mod.rs (hunk 2 above)
Comment (connormullett) [reply ipld/src/selector/mod.rs]: 👍

File: .config/forest.dic (hunk 4 above)
Comment (connormullett) [reply .config/forest.dic]: 👍

File: .config/forest.dic (hunk 6 above)
Comment (lemmih) [reply .config/forest.dic]: Can we simplify the wording of the documentation to avoid adding this word?

File: .config/forest.dic [hunk 7]
```diff
@@ -31,15 +31,16 @@ tipset/SM
 tipsetkey/S
 tuple
 validator/S
-serializing
-deserializing
 mempool
 sortition
+serializer/SM
+deserialize/DG
```
Comment (elmattic) [inline .config/forest.dic:37]: Consider using `de-serialize`/ `de-serializer` in our codebase so we don't need to add those in `forest.dic`.

**Review by elmattic — CHANGES_REQUESTED** [review]

File: .config/forest.dic (hunk 5 above)
Comment (connormullett) [reply .config/forest.dic:36]: Removing serializer from forest.dic causes the errors to reappear, doesn't look like it is in the dictionary?

File: .config/forest.dic (hunk 6 above)
Comment (connormullett) [reply .config/forest.dic]: Sure!

File: .config/forest.dic (hunk 7 above)
Comment (connormullett) [reply .config/forest.dic]: Done

File: .config/forest.dic (hunk 6 above)
Comment (connormullett) [reply .config/forest.dic]: Reworded the conflicting documentation from "foo/bar/baz" to "path/to/file"

File: .config/forest.dic (hunk 5 above)
Comment (lemmih) [reply .config/forest.dic:36]: Merge with `main` and try again.

File: ipld/src/selector/mod.rs [hunk 8]
```diff
@@ -168,13 +168,13 @@ impl SubAssign<u64> for RecursionLimit {
...
-///   - in ExploreRecursive, to halt exploration.
-///   - in ExploreConditional,
+///   - in `Matcher`, to determine if a node is selected.
+///   - in `ExploreRecursive`, to halt exploration.
+///   - in `ExploreConditional`,
 ///
 ///
-/// TODO -- Condition is very skeletal and incomplete.
-/// The place where Condition appears in other structs is correct;
+/// Condition is very skeletal and incomplete.
```
Comment (lemmih) [inline ipld/src/selector/mod.rs:176]: The TODO comment is valid and shouldn't be removed. It should just be changed from a documentation comment to a normal comment.

Comments that start with three slashes are used for documentation. Comments that start with 2 slashes are used for internal development. TODO comments are internal and should never be part of the documentation.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: .config/forest.dic (hunk 5 above)
Comment (connormullett) [reply .config/forest.dic:36]: Tried again, same thing

File: ipld/src/selector/mod.rs (hunk 8 above)
Comment (connormullett) [reply ipld/src/selector/mod.rs]: Re-added

File: .config/forest.dic (hunk 5 above)
Comment (lemmih) [reply .config/forest.dic:36]: Ah, I saw it wrong, only `serialize` is in the dictionary.

**Review by lemmih — APPROVED** [review]

**Review by jdjaustin — APPROVED** [review]
